commit/metrics/prom.go

```diff
@@ -0,0 +1,224 @@
+			Help: "This metric tracks the number of different items in the merkle observation of the commit plugin",
+		},
+		[]string{"chainID", "state", "type"},
+	)
+	promMerkleOutcomeDetails = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ccip_commit_merkle_processor_outcome_components_sizes",
+			Help: "This metric tracks the number of different items in the merkle outcome of the commit plugin",
+		},
+		[]string{"chainID", "state", "type"},
+	)
+	promTokenPriceObservationDetails = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ccip_commit_token_processor_observation_components_sizes",
+			Help: "This metric tracks the number of different items in the token prices observation of the commit plugin",
+		},
+		[]string{"chainID", "type"},
+	)
+	promTokenPriceOutcomeDetails = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ccip_commit_token_processor_outcome_components_sizes",
+			Help: "This metric tracks the number of different items in the token prices outcome of the commit plugin",
+		},
+		[]string{"chainID", "type"},
+	)
+	promChainFeeObservationDetails = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ccip_commit_chain_fee_processor_observation_components_sizes",
+			Help: "This metric tracks the number of different items in the chain fee observation of the commit plugin",
+		},
+		[]string{"chainID", "type"},
+	)
+	promChainFeeOutcomeDetails = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ccip_commit_chain_fee_processor_outcome_components_sizes",
+			Help: "This metric tracks the number of different items in the chain fee outcome of the commit plugin",
+		},
+		[]string{"chainID", "type"},
+	)
+)
+
+type PromReporter struct {
+	lggr    logger.Logger
+	chainID string
+	// Prometheus components
+	merkleProcessorObservationCounter   *prometheus.CounterVec
+	merkleProcessorOutcomeCounter       *prometheus.CounterVec
+	tokenProcessorObservationCounter    *prometheus.CounterVec
+	tokenProcessorOutcomeCounter        *prometheus.CounterVec
+	chainFeeProcessorObservationCounter *prometheus.CounterVec
+	chainFeeProcessorOutcomeCounter     *prometheus.CounterVec
+}
+
+func NewPromReporter(lggr logger.Logger, selector cciptypes.ChainSelector) (*PromReporter, error) {
+	chainID, err := sel.GetChainIDFromSelector(uint64(selector))
+	if err != nil {
+		return nil, err
+	}
+
+	return &PromReporter{
+		lggr:    lggr,
+		chainID: chainID,
+
+		merkleProcessorObservationCounter:   promMerkleProcessorObservationDetails,
+		merkleProcessorOutcomeCounter:       promMerkleOutcomeDetails,
+		tokenProcessorObservationCounter:    promTokenPriceObservationDetails,
+		tokenProcessorOutcomeCounter:        promTokenPriceOutcomeDetails,
+		chainFeeProcessorObservationCounter: promChainFeeObservationDetails,
+		chainFeeProcessorOutcomeCounter:     promChainFeeOutcomeDetails,
+	}, nil
+}
+
+func (p *PromReporter) TrackObservation(obs committypes.Observation) {
```

Comment:
To be implemented later, I want to check metrics first on staging before making a decision whether to keep it or not